orangewidget/widget.py

```diff
@@ -383,34 +380,15 @@ def _insert_splitter(self):
         self.__splitter = self._Splitter(Qt.Horizontal, self)
         self.layout().addWidget(self.__splitter)
 
-    class _HiddenVerticalScrollArea(VerticalScrollArea):
-
-        def __init__(self, parent):
-            super().__init__(parent)
-            self.setFrameShape(QFrame.NoFrame)
-
-        def sizeHint(self):
-            if self.widget():
-                return self.widget().sizeHint()
-            else:
-                return super().sizeHint()
-
-        def minimumSizeHint(self):
-            if self.widget():
-                min_height = min(self.widget().minimumSizeHint().height(), 300)
-                return QSize(self.widget().minimumSizeHint().width(), min_height)
-            else:
-                return super().minimumSizeHint()
-
     def _insert_control_area(self):
-        if self.left_side_scrolling:
-            scroll_area = self._HiddenVerticalScrollArea(self)
-            self.__splitter.addWidget(scroll_area)
-            self.left_side = gui.vBox(scroll_area, spacing=0)
-            scroll_area.setWidget(self.left_side)
-        else:
-            self.left_side = gui.vBox(self.__splitter, spacing=0)
-        self.__splitter.setSizes([1])  # Smallest size allowed by policy
+        scroll_area = VerticalScrollArea(self)
+        self.__splitter.addWidget(scroll_area)
+        self.left_side = gui.vBox(scroll_area, spacing=0,
+                                  sizePolicy=(QSizePolicy.Fixed,
+                                              QSizePolicy.MinimumExpanding))
+        scroll_area.setWidget(self.left_side)
+        scroll_area.setMinimumWidth(1)  # forces it not to stretch on initial show
```

Comment:
I assume the 
```python
self.__splitter.setSizes([1])  # Smallest size allowed by policy
``` 
line was here to make the left side not stretch when the window is resized horizontally. This worked for all widgets except Heat Map, for some reason. However, it worked correctly after collapsing and uncollapsing the left area. `scroll_area.setMinimumWidth(1)` seems to replace this just fine, but perhaps I didn't test it with enough widgets to make sure it's the definitive solution.